### PR TITLE
fix: revoke not handled with replication :beetle:

### PR DESCRIPTION
**Pull request by enguerran**
The message between the automatic pouchDB replication
was not handled well by the application.

### CODE REVIEW

File: mobile/src/main.jsx [hunk 1]
```diff
@@ -37,7 +37,7 @@ const renderAppWithPersistedState = persistedState => {
           cozy.client.offline.startRepeatedReplication('io.cozy.files', 15, options)
           initBar()
         } else {
-          onError(store.dispatch, store.getState)()
+          onError(store.dispatch, store.getState)({ message: 'Client has been revoked' })
```
Comment (kosssi) [inline mobile/src/main.jsx]: maybe you can use https://github.com/cozy/cozy-files-v3/pull/207/files#diff-f96699990a60f265f8d3b7bd25d3e9a6R8 ?

File: mobile/src/lib/cozy-helper.js [hunk 2]
```diff
@@ -83,10 +83,10 @@ export function refreshFolder (dispatch, getState) {
 }
 
 export const onError = (dispatch, getState) => (err) => {
-  if (err.message === clientRevokedMsg) {
+  if (err.message === clientRevokedMsg || err.error === 'code=400, message=Invalid JWT token') {
```
Comment (kosssi) [inline mobile/src/lib/cozy-helper.js]: it's not the same here https://github.com/cozy/cozy-files-v3/pull/207/files#diff-f96699990a60f265f8d3b7bd25d3e9a6R51 ?

File: mobile/src/lib/cozy-helper.js [hunk 3]
```diff
@@ -83,10 +83,10 @@ export function refreshFolder (dispatch, getState) {
...
     console.warn(`Your device is no more connected to your server: ${getState().mobile.settings.serverUrl}`)
     dispatch(revokeClient())
-  } else if (err.message === 'ETIMEDOUT' || err.message === 'getCheckpoint rejected with ') {
+  } else if (err.message === 'ETIMEDOUT') {
```
Comment (kosssi) [inline mobile/src/lib/cozy-helper.js]: why you remove this?

Comment (enguerran) [inline mobile/src/lib/cozy-helper.js]: It appears that pouchDB replication encapsulates network error in an error with message `getCheckpoint rejected with`. For instance when your token is not correct, cozy-stack responds with a `Invalid JWT Token` translated by cozy-client-js to `Client has been revoked`. On the other hand, the pouchDB replication encapsulates the `Invalid JWT Token` in an error with `getCheckpoint rejected with` as the message.
So it is not very accurate as it catches others errors than `timeout` errors and I prefer to remove this and use `err.error` to see what's the real error message is.

File: mobile/src/lib/cozy-helper.js (hunk 2 above)
Comment (enguerran) [inline mobile/src/lib/cozy-helper.js]: PouchDB encapsulates network error message in another error with a different generic message. The real error is available in `err.error`.

File: mobile/src/main.jsx (hunk 1 above)
Comment (enguerran) [inline mobile/src/main.jsx]: I write that code this way as it brings to light the need of a refactoring. It is a big smell and I would like to fix the bug, not to refactor the whole error management in pouchDB replication.

This is not an easy move as it involves a lot of different actors:
- the different cozy-stack's error ;
- the cozy-client-js layer ;
- the pouchDB internal layer ;
- the cozy-files-v3 handling those all together.

It worth to be refactored once and for all now we have all the actors in the place, imho.

**Pull request merged**